Makefile

```diff
@@ -41,7 +41,11 @@ PROTOC=protoc
 
 # https://docs.docker.com/build/building/multi-platform/
 # Make sure to always build containers using AMD64 but allow to be overridden by users if need for cross-os compatibility.
-CONTAINER_ARCH=linux/amd64
+CONTAINER_OS_ARCH=linux/amd64
+# Allow to independently customise go OS and ARCH flags independently and following the standard practices.
+# Defaulting to linux/amd64 as per CONTAINER_OS_ARCH.
+GOOS=linux
```

Comment:
this looks like it could cause silent issues for ppl building the binaries to just run them locally. I think it would be better to just set them to `$(shell go env GOOS)` and if someone wants to override them via a environment variable, they can